pytket/phir/sharding/shards2ops.py

```diff
@@ -39,13 +43,23 @@ def parse_shards_naive(
             # if there are more than 2 qubits used, treat them all as parallel sq ops
             # one qubit will just be a single sq op
             # 3 or more will be 3 or more parallel sq ops
+            # when iterating through qubits,
+            # map all the qubits to a unique id to prevent duplicates in placement
             if len(shard.qubits_used) != 2:  # noqa: PLR2004
                 for qubit in shard.qubits_used:
-                    op = qubit.index
+                    if qubit not in qubits2ids:
+                        qubits2ids[qubit] = qubit_id
+                        qubit_id += 1
+                    qid = qubits2ids[qubit]
+                    op = [qid]
                     layer.append(op)
             else:
                 for qubit in shard.qubits_used:
-                    op.append(qubit.index[0])
+                    if qubit not in qubits2ids:
+                        qubits2ids[qubit] = qubit_id
+                        qubit_id += 1
+                    qid = qubits2ids[qubit]
```

Comment:
Could you write a helper function to avoid duplication of these 4 lines of code in the two branches?